tuf/tuf.go

```diff
@@ -202,6 +223,47 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey, before
 
 	tr.keysDB.AddRole(role)
 
+	utils.RemoveUnusedKeys(p)
+
+	return nil
+}
+
+// DeleteDelegation removes a delegated targets role from its parent
+// targets object. It also deletes the delegation from the snapshot.
+// DeleteDelegation will only make use of the role Name field.
+func (tr *Repo) DeleteDelegation(role data.Role) error {
+	if !role.IsDelegation() {
+		return data.ErrInvalidRole{Role: role.Name, Reason: "not a valid delegated role"}
+	}
+	// the role variable must not be used past this assignment for safety
+	name := role.Name
+
+	parent := filepath.Dir(name)
+	p, ok := tr.Targets[parent]
+	if !ok {
+		return data.ErrInvalidRole{Role: name, Reason: "parent role not found"}
```

Comment:
It's a very esoteric difference. For the purposes of delete I could probably just return nil in the case the parent doesn't exist. The error though will help us behave correctly in the case of failures where the parent role hasn't been loaded (via logic errors or otherwise).